### ISSUE TITLE
Multi-line doc comment doesn't reformat the whole paragraph

### ISSUE THREAD

**Issue by turboladen** [rust-lang-nursery/rustfmt#1285]
I poked through many of the existing "comments"-related issues, but didn't land on something that fit this bill, so here it is:

```
❯ rustc --version
rustc 1.16.0-nightly (1a2ed98d3 2017-01-13)
❯ rustfmt --version
0.7.1 ()
```

My `rustfmt.toml`:

```toml
normalize_comments = true
reorder_imported_names = true
reorder_imports = true
report_todo = "Always"
wrap_comments = true
```

Before:

```rust
/// Per unitwise:
///
/// A `Unit` is a collection of `Term`s. `Term`s can be combined via multiplication
/// or division to create a unit. A `Unit` does not have a magnitude, but does
/// have a scale.
///
```

After:

```rust
/// Per unitwise:
///
/// A `Unit` is a collection of `Term`s. `Term`s can be combined via
/// multiplication
/// or division to create a unit. A `Unit` does not have a magnitude, but does
/// have a scale.
///
```

So that first line of the paragraph hits a line-length limit and properly gets pushed to the next line, but subsequent lines don't get joined to reformat the paragraph. If it's too much to redo the whole paragraph, I'd be fine with just getting a warning.

----

FWIW, I'm coming from the Ruby world and use [rubocop](https://github.com/bbatsov/rubocop) for some static analysis stuff. I think it's focus is primarily the analysis & feedback, but secondarily it will reformat your code (like rustfmt). They add new warnings for ugly code before they write code that'll fix those things; this is kinda nice because it at least lets me go manually fix some things.

Before rubocop was around, I wrote my own Rust static analysis tool--it was great for some things, but turned out to be a handful to maintain; I noticed you've got a lot of comments-related issues, so in the case where you're falling behind in getting to them, I just thought I'd throw out the suggestion that if it's any easier, you might take a similar approach to rubocop in that reporting and fixing are separate things (assuming it's more work to fix bad formatting than it is to detect bad formatting).

Anyway, just a thought. I realize that rust and rustfmt are a different world, so maybe this suggestion doesn't make sense here. In any case, thank you for the great tool!

**Comment by topecongiro** [comment]
Fixed on `0.2.16-nightly`, closing.

**Comment by turboladen** [comment]
💫
